extensions/amazon-sam/runtime/src/main/java/io/quarkus/amazon/sam/SamHandler.java

```diff
@@ -0,0 +1,192 @@
+            try {
+                if (msg == null)
+                    continue;
+
+                if (msg instanceof HttpResponse) {
+                    HttpResponse res = (HttpResponse) msg;
+                    response = new AwsProxyResponse(res.status().code());
+                    for (Map.Entry<String, String> entry : res.headers()) {
+                        response.addHeader(entry.getKey(), entry.getValue());
+                    }
+                }
+                if (msg instanceof HttpContent) {
+                    HttpContent content = (HttpContent) msg;
+                    if (baos == null) {
+                        // todo what is right size?
+                        baos = new ByteArrayOutputStream(500);
+                    }
+                    int readable = content.content().readableBytes();
+                    for (int i = 0; i < readable; i++) {
+                        baos.write(content.content().readByte());
+                    }
+                }
+                if (msg instanceof LastHttpContent) {
+                    response.setBody(new String(baos.toByteArray(), StandardCharsets.UTF_8));
+                    return response;
+                }
+            } finally {
+                if (msg != null)
+                    ReferenceCountUtil.release(msg);
+            }
+        }
+    }
+
+    private int getPort(AwsProxyRequest request) {
+        if (request.getMultiValueHeaders() == null) {
+            return 443;
+        }
+        String port = request.getMultiValueHeaders().getFirst("X-Forwarded-Port");
+        return SecurityUtils.isValidPort(port) ? Integer.parseInt(port) : 443;
+    }
+
+    private DefaultHttpRequest buildRequest(AwsProxyRequest request) {
+
+        String path = request.getPath();
+        String query = request.getQueryString();
+        if (query != null && !"".equals(query.trim())) {
+            path = path + '?' + query;
+        }
+        String host = request.getMultiValueHeaders().getFirst(HttpHeaders.HOST);
+        if (host.matches(".*:[0-9]+")) {
+            host = host.substring(0, host.indexOf(':'));
+        }
+        int port = getPort(request);
+        if (port != -1) {
+            host = host + ':' + port;
+        }
+        DefaultHttpRequest nettyRequest = new DefaultHttpRequest(HttpVersion.HTTP_1_1,
+                HttpMethod.valueOf(request.getHttpMethod()), path);
+        nettyRequest.headers().set("Host", host);
+        for (Map.Entry<String, List<String>> header : request.getMultiValueHeaders().entrySet()) {
+            nettyRequest.headers().add(header.getKey(), header.getValue());
+        }
+
+        return nettyRequest;
+    }
+
+    private HttpContent buildContent(AwsProxyRequest request) {
+        String body = request.getBody();
+        return body != null
+                ? new DefaultLastHttpContent(wrappedBuffer(body.getBytes(StandardCharsets.UTF_8)))
```

Comment:
Again, you do not handle non-text types.  See stuarts implementation of this:

https://github.com/quarkusio/quarkus/blob/c58a1c1eace83126de787b5003f11a619227500c/extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/runtime/AwsHttpHandler.java#L82